src/Analyzers/CSharp/Analyzers/RemoveUnreachableCode/RemoveUnreachableCodeHelpers.cs

```diff
@@ -55,37 +69,87 @@ public static ImmutableArray<ImmutableArray<StatementSyntax>> GetSubsequentUnrea
 
         var sections = ArrayBuilder<ImmutableArray<StatementSyntax>>.GetInstance();
 
-        var currentSection = ArrayBuilder<StatementSyntax>.GetInstance();
-        var firstUnreachableStatementIndex = siblingStatements.IndexOf(firstUnreachableStatement);
+        var firstStatementSpan = firstUnreachableStatement.Span;
 
-        for (int i = firstUnreachableStatementIndex + 1, n = siblingStatements.Length; i < n; i++)
+        var currentSection = ArrayBuilder<StatementSyntax>.GetInstance();
+        var firstUnreachableStatementIndex = siblingStatements.IndexOf(s => s.Span.Contains(firstStatementSpan));
```

Comment:
i don't undrestand this.

Reply:
The first unreachable statement may be contained inside a nested block, and in the previous part of the code we have calculated the control flow scope's siblings off of the outermost block that directly contains blocks. To avoid traversing the descendants and comparing against them, we use a span-based comparison to determine which block contains the first unreachable statement.